scripts/import_geometries.py

```diff
@@ -4,319 +4,381 @@
     return inserted
 
+
 # Main execution
-try:
-    # Recreate table without the constraint if needed
-    print("Checking if we need to recreate the table without the constraint...")
-    constraints = con.execute("""
-    SELECT sql FROM sqlite_master 
-    WHERE type='table' AND name='precinct_geometries'
-    """).fetchone()
-    
-    if constraints and 'UNIQUE' in constraints[0]:
-        print("Found UNIQUE constraint, recreating table without it...")
-        
-        # Create a backup table
-        con.execute("CREATE TABLE IF NOT EXISTS precinct_geometries_backup AS SELECT * FROM precinct_geometries")
-        print("Created backup of existing data")
-        
-        # Drop the original table
-        con.execute("DROP TABLE precinct_geometries")
-        print("Dropped original table")
-        
-        # Create a new table without the constraint
-        con.execute("""
-        CREATE TABLE IF NOT EXISTS precinct_geometries (
-            precinct_geometry_id INTEGER PRIMARY KEY,
-            precinct_id VARCHAR,
-            valid_from_year INTEGER,
-            valid_to_year INTEGER,
-            geometry GEOMETRY
-        );
-        """)
-        print("Created new table without UNIQUE constraint")
-        
-        # Restore data if backup exists
-        backup_count = con.execute("SELECT COUNT(*) FROM precinct_geometries_backup").fetchone()[0]
-        if backup_count > 0:
-            con.execute("""
-            INSERT INTO precinct_geometries
-            SELECT * FROM precinct_geometries_backup
-            """)
-            print(f"Restored {backup_count} records from backup")
-    else:
-        print("Table has no UNIQUE constraint or doesn't exist yet, proceeding...")
-    
-    # Import each GeoJSON with its valid year range
-    total_imported = 0
-    
-    # Import files one at a time, committing after each one
-    print("\n--- Importing 2010 precincts ---")
-    con.execute("BEGIN TRANSACTION")
-    total_imported += import_precinct_geojson(
-        '/home/andrew/projects/cgp_election_data/precincts/2010_precincts.geojson', 
-        2010, 2013
-    )
-    con.execute("COMMIT")
-    
-    print("\n--- Importing 2014 precincts ---")
-    con.execute("BEGIN TRANSACTION")
-    total_imported += import_precinct_geojson(
-        '/home/andrew/projects/cgp_election_data/precincts/2014_precincts.geojson', 
-        2014, 2021
-    )
-    con.execute("COMMIT")
-    
-    print("\n--- Importing 2022 precincts ---")
-    con.execute("BEGIN TRANSACTION")
-    total_imported += import_precinct_geojson(
-        '/home/andrew/projects/cgp_election_data/precincts/2022_precincts.geojson', 
-        2022, None
+if __name__ == "__main__":
+    con.execute(
+        """
+    CREATE TABLE IF NOT EXISTS precinct_geometries (
+        precinct_geometry_id INTEGER PRIMARY KEY,
+        precinct_id VARCHAR,
+        valid_from_year INTEGER,
+        valid_to_year INTEGER,
+        geometry GEOMETRY
+    );
+    """
```

Comment:
Basically I noticed that if you don't run this outside of the try block, the script will fail and claim this table doesn't exist. So you have to create it first (I suppose you could do it in the try block too).